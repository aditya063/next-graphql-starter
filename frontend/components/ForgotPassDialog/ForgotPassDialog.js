import Modal from '../Modal/Modal';
import * as sc from './ForgotPassDialog.style';

class ForgotPassDialog extends React.PureComponent {
  state = { modal: '' };

  render() {
    return (
      <>
        <sc.ForgotPassDialog
          value="Forgot password?"
          onClick={() => this.setState({ modal: 'requestReset' })}
        >
          Forgot password?
        </sc.ForgotPassDialog>

        {this.state.modal && (
          <Modal
            modal={this.state.modal}
            setModal={modalName => this.setState({ modal: modalName })}
          />
        )}
      </>
    );
  }
}

export default ForgotPassDialog;
