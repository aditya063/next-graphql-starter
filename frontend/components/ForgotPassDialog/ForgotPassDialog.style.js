import styled from 'styled-components';
import { Modal, Button } from 'antd';

export const ForgotPassDialog = styled.div``;

export const ModalLink = styled.a``;

export const ForgotPassModal = styled(Modal)``;

export const SubmitBtn = styled(Button)``;
