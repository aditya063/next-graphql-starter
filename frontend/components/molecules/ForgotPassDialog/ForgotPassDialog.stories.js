import ForgotPassDialog from './ForgotPassDialog';

export default { title: 'Molecules', component: ForgotPassDialog };

export const forgotPassDialog = () => <ForgotPassDialog />;
