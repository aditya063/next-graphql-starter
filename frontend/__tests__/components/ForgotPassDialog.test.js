import {
  prettyDOM,
  render,
  screen,
  fireEvent,
  wait
} from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import ForgotPassDialog from '../../components/ForgotPassDialog/ForgotPassDialog';
// import RequestReset from '../../components/RequestReset/RequestReset';
import theme from '../../public/styles/theme.style';

// jest.mock('../../components/RequestReset/RequestReset', () => {
//   const RequestReset = () => 'RequestReset';
//   return RequestReset;
// });

jest.mock('../../components/RequestReset/RequestReset', () => () =>
  'RequestReset'
);

// jest.mock(RequestReset, () => {
//   const RequestReset = () => <div />;
//   return { __esModule: true, default: RequestReset };
// }).default;

// jest.mock('../../components/RequestReset/RequestReset', () => {
//   const RequestReset = <div />;
//   return RequestReset;
// });

describe('ForgotPassDialog', () => {
  it('matches snapshot', () => {
    const mockProps = {};

    const utils = render(
      <ThemeProvider theme={theme}>
        <ForgotPassDialog {...mockProps} />
      </ThemeProvider>
    );

    const container = utils.container.firstChild;

    expect(container).toMatchSnapshot();
  });

  it('matches snapshot', () => {
    const mockProps = {};

    const utils = render(
      <ThemeProvider theme={theme}>
        <ForgotPassDialog {...mockProps} />
      </ThemeProvider>
    );

    const container = utils.container.firstChild;

    // console.log('TCL: container', prettyDOM(container));
    // console.log('TCL: container1', prettyDOM(utils.baseElement));

    const modalLink1 = utils.getByText('Forgot password?');
    console.log('TCL: modalLink1', prettyDOM(modalLink1));

    const modalLink2 = screen.getByText('Forgot password?');
    console.log('TCL: modalLink2', prettyDOM(modalLink2));

    // fireEvent.click(modalLink);

    // console.log('TCL: container2', prettyDOM(utils.baseElement));

    // expect(container).toHaveTextContent(footerText);
  });
});
