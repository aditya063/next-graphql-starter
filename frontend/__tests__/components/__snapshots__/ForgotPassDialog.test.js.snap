// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPassDialog matches snapshot 1`] = `
<div
  class=""
>
  <a
    class=""
  >
    Forgot password?
  </a>
</div>
`;
